### PR TITLE
Fix validation for autoscaler deployment

### LINKED ISSUE #1397: Cluster creation w/o Node AutoScaling still checks for autoscaler
**Bug description**
I created a workload cluster without autoscaling.  During the creation, output suggests it was looking for autoscaler and took about 2minutes before it gave-up because it could not be found and continued with the deployment.

```
$ tanzu cluster create -f manifests/clusters/platform-sandbox-2/cluster-config.yaml --tkr v1.21.6---vmware.1-tkg.1-rc.3 
Validating configuration...
Creating workload cluster 'platform-sandbox-2'...
Waiting for cluster to be initialized...
cluster control plane is still being initialized: ScalingUp
Waiting for cluster nodes to be available...
Waiting for cluster autoscaler to be available...
Unable to wait for autoscaler deployment to be ready. reason: deployments.apps "platform-sandbox-2-cluster-autoscaler" not found
Waiting for addons installation...
Waiting for packages to be up and running...
...
```
**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [ ] Addons
- [x] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Expected behavior**
Since I did not configure with autoscaling, I would not expect the deployment to seek autoscaler and increase deployment time looking for a phantom autoscaler. 

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

### PR DESCRIPTION
**Pull request by saimanoj01**
### What this PR does / why we need it
Skip checking for autoscaler deployment, if cluster is created without autoscaler.

### Which issue(s) this PR fixes

Fixes #1397

### Describe testing done for PR
1. Created a cluster by setting 'ENABLE_AUTOSCALER=false' and made sure that the CLI does not wait for autoscaler.
2. Created a cluster by setting 'ENABLE_AUTOSCALER=true' and made sure that the CLI waits for autoscaler.

### Release note

```release-note
Skip checking for autoscaler deployment, if cluster is created without autoscaler.
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/tkg/client/create_test.go [hunk 1]
```diff
@@ -115,4 +116,36 @@ spec:
 			})
 		})
 	})
+
+	Describe("WaitForAutoscalerDeployment", func() {
```
Comment (swalner-vmware) [inline pkg/v1/tkg/client/create_test.go:120]: Nice to have: a test where the TKGConfigReaderWriter returns an error

**Review by swalner-vmware — APPROVED** [review]
LGTM; nice tests

File: pkg/v1/tkg/client/create_test.go (hunk 1 above)
Comment (saimanoj01) [reply pkg/v1/tkg/client/create_test.go:120]: This is the case where TKGConfigReaderWriter returns an error.

https://github.com/vmware-tanzu/tanzu-framework/pull/1398/files#diff-7d1cc721897f32ac26349f1835174055ad258f3665adee85734b23f3f024e269R125

**Review by saji-pivotal — APPROVED** [review]

File: pkg/v1/tkg/client/cluster.go [hunk 2]
```diff
@@ -182,13 +182,7 @@ func (c *TkgClient) waitForClusterCreation(regionalClusterClient clusterclient.C
...
 	}
 
-	if _, err := c.TKGConfigReaderWriter().Get(constants.ConfigVariableEnableAutoscaler); err == nil {
-		log.Infof("Waiting for cluster autoscaler to be available...")
-		autoscalerDeploymentName := options.ClusterName + "-cluster-autoscaler"
-		if err := regionalClusterClient.WaitForAutoscalerDeployment(autoscalerDeploymentName, options.TargetNamespace); err != nil {
-			log.Warningf("Unable to wait for autoscaler deployment to be ready. reason: %v", err)
-		}
-	}
+	c.WaitForAutoscalerDeployment(regionalClusterClient, options.ClusterName, options.TargetNamespace)
```
Comment (imikushin) [inline pkg/v1/tkg/client/cluster.go:185]: **praise**: Extracted method helps make the code simpler and easier to understand.

File: pkg/v1/tkg/client/cluster.go [hunk 3]
```diff
@@ -214,6 +208,30 @@ func (c *TkgClient) waitForClusterCreation(regionalClusterClient clusterclient.C
...
+
+	if autoscalerEnabled, err = c.TKGConfigReaderWriter().Get(constants.ConfigVariableEnableAutoscaler); err != nil {
+		return
+	}
+
+	var isEnabled bool
+	if isEnabled, err = strconv.ParseBool(autoscalerEnabled); err != nil {
+		log.Warningf("Unable to parse the value of config variable %q. reason: %v", constants.ConfigVariableEnableAutoscaler, err)
+		return
+	}
```
Comment (imikushin) [inline pkg/v1/tkg/client/cluster.go:224]: This can be another helper method that returns `bool`.

Comment (saimanoj01) [reply pkg/v1/tkg/client/cluster.go]: Done!

Comment (imikushin) [reply pkg/v1/tkg/client/cluster.go]: Sorry, I should have made my point more clear. I was talking about the whole piece of code beginning with `var autoscalerEnabled string` declaration. Also, just noticed, we're swallowing the error doing this:
```go
	if autoscalerEnabled, err = c.TKGConfigReaderWriter().Get(constants.ConfigVariableEnableAutoscaler); err != nil {
		return
	}
```

**Review by anujc25 — APPROVED** [review]
Looks good

**Pull request merged**